src/AggregateFunctions/AggregateFunctionGroupSortedArrayData.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+#include <IO/VarInt.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteHelpers.h>
+
+#define DEFAULT_THRESHOLD 10
+
+namespace DB
+{
+template <typename Storage>
+struct AggregateFunctionGroupSortedArrayDataBase
+{
+    typedef typename Storage::value_type ValueType;
```

Comment:
This is the key, 'typename Storage' defines what kind of std storage we are using, so we define here a std::multiset<X>::value_type or std::multimap<X,Y>::value_type, so here we operate with this abstraction.